contracts/Loan.sol

```diff
@@ -0,0 +1,407 @@
+pragma solidity 0.4.24;
+
+import "./LoanFactory.sol";
+import "./lib/SafeMath.sol";
+
+contract Loan {
+    using SafeMath for uint256;
+
+    enum LoanStatus {
+        Pending,
+        Active,
+        InterestPayment,
+        Liquidated,
+        Matured,
+        Completed
+    }
+
+    struct Lender {
+        bytes32 id;
+        bytes32 orderId;
+        bytes32 lenderUserId;
+        uint256 amount;
+        uint256 rate;
+        uint256 amountWeight;
+        uint256 rateWeight;
+    }
+
+    struct Interest {
+        uint256 paymentTime;
+        uint256 amount;
+        bool paid;
+    }
+
+    LoanFactory public loanFactory;
+    uint256 public tenor;
+    uint256 public principalAmount;
+    uint256 public collateralAmount;
+    uint256 public createdTime;
+    uint256 public lowerRequiredMargin;
+    uint256 public higherRequiredMargin;
+    uint256 public lastMarginTime;
+    uint256 public constant WEIGHT_DIVISOR = 10000;
+    bytes32 public constant INTEREST_CURRENCY = 0x4c4e44;
+    bytes32 public borrowerUserId;
+    bytes32 public holdingUserId;
+    bytes32 public escrowUserId;
+    bytes32 public liquidatorUserId;
+    bytes32 public id;
+    bytes32 public orderId;
+    bytes32 public principalCurrency;
+    bytes32 public collateralCurrency;
+    LoanStatus public status;
+    Lender[] public lenders;
+    Interest[] public interest;
+    uint256 public totalExpectedTransfers;
+    bytes32[] public transferOutcomeRecords;
+    bytes32 public constant TRANSFER_NOT_SENT = 0x4e4f545f53454e54;
+
+    event ExpectedTransfer(bytes32 from, bytes32 to, uint256 amount, bytes32 currency, uint256 totalExpectedTransfers, string functionName);
+
+    modifier onlyOwner() {
+        require(msg.sender == loanFactory.owner());
+        _;
+    }
+
+    modifier onlyWorker() {
+        require(msg.sender == loanFactory.worker());
+        _;
+    }
+
```

Comment:
may be we should put modifiers outside of `Loan` ideally in a separate folder or file

Reply:
have another contract just with modifiers? I don't that's the recommended way